### PR TITLE
Added checkfile for unixfilesystem free space (main)

### PR DESCRIPTION
**Pull request by ganning127**
**************************************************************************************************Invalid free space values & less than 10 GB free:**************************************************************************************************

image

******************************************Less than 10 GB free:******************************************

image

******Invalid free space values:******

image

**********************All correct**********************

image

### CODE REVIEW

File: src/data/checkfiles/default_checkfiles/unixfilesystem-resc-space.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
+import { Link } from '@reach/router'
+import React from 'react'
+
+export default {
+    name: "unixfilesystem resources have at least 10 GB of free space",
+    description: "Checks if all unixfilesystem resources have at least 10 GB of free space",
+    minimum_server_version: "4.2.0",
+    maximum_server_version: "",
+    interval_in_seconds: 3600,
```
Comment (trel) [inline src/data/checkfiles/default_checkfiles/unixfilesystem-resc-space.js:9]: let's bump this up to ... daily?    so 86400

**Review by trel — COMMENTED** [review]
let's show the 'too small' results as 'xxxx (y.y GB)'

where xxxx is the `freeSpaceInBytes`, and the y.y is `freeSpaceInGB`.

**Comment by ganning127** [comment]
Hmm, that's what this would look like:

Screenshot 2023-06-22 at 12 09 00 AM

**Comment by trel** [comment]
well... I don't hate it, but don't love it either.

separately, can we pull out the '10' that is used in a number of places to a single variable?

**Comment by ganning127** [comment]
Same, I'll brainstorm a couple of options on formatting.

I pulled out the 10 GB into a separate variable, but I don't think we can change it in the `name` and `description`. I tried setting a property and referencing that property in `name` and `description`, but no luck. https://stackoverflow.com/questions/4618541/can-i-reference-other-properties-during-object-declaration-in-javascript

File: src/data/checkfiles/default_checkfiles/unixfilesystem-resc-space.js [hunk 2]
```diff
@@ -2,7 +2,7 @@ import { Link } from '@reach/router'
 import React from 'react'
 
 export default {
-    name: "unixfilesystem resources have at least 10 GB of free space",
+    name: `unixfilesystem resources have at least 10 GB of free space`,
```
[the remark this answers is not in the document]
Comment (ganning127) [reply src/data/checkfiles/default_checkfiles/unixfilesystem-resc-space.js:5]: was playing around with templating a variable for 10GB into there but forgot to change it back

File: sample.env [hunk 3]
```diff
@@ -14,3 +14,4 @@ REACT_APP_APPBAR_LOGO=iRODS-logo.jpg
 REACT_APP_LOGIN_LOGO=iRODS-logo-1.png
 REACT_APP_BRANDING_NAME=Zone Management Tool
 REACT_APP_PRIMARY_COLOR=#04bdaf
+REACT_APP_MIN_GB=10
```
Comment (trel) [inline sample.env:17]: this feels... almost too high level... can we just make it a variable in the checkfile itself?  same as the 'interval_in_seconds'.   this name doesn't have meaning across the entire app.

how about... `minimum_size_in_gb`... at top of unixfilesystem-resc-space.js.

Comment (ganning127) [reply sample.env:17]: ohhh yep, i really like that idea

**Comment by ganning127** [comment]
We could also show the not enough space like this, though I'm not sure if it gets confusing saying we're dividing those bytes by that GB value:

Screenshot 2023-06-23 at 9 55 27 AM

**Comment by trel** [comment]
right, that presentation looks like math... hmmm

**Comment by ganning127** [comment]
Some ideas:
- (23242842 bytes, or 8 GB)
- (23242842 bytes, or 8 GB remaining)
- (23242842 bytes = 8 GB) # looks like a conversation chart to me, don't like this one too much

**Comment by trel** [comment]
oh oh, mouseover?   tool tip?

so...

'rescName (y.y GB)' with tooltip that shows 'xxxx bytes'

**Comment by ganning127** [comment]
ohh I like that. Here's what it looks like (when hovering over the`(0.8 GB)`:

Screenshot 2023-06-23 at 1 51 49 PM

**Comment by trel** [comment]
yes please, this is good.  squash, no #s!

**Comment by ganning127** [comment]
done!

File: sample.env [hunk 4]
```diff
@@ -13,4 +13,4 @@ LISTEN_PORT=9000
 REACT_APP_APPBAR_LOGO=iRODS-logo.jpg
 REACT_APP_LOGIN_LOGO=iRODS-logo-1.png
 REACT_APP_BRANDING_NAME=Zone Management Tool
-REACT_APP_PRIMARY_COLOR=#04bdaf
+REACT_APP_PRIMARY_COLOR=#04bdaf
```
Comment (trel) [inline sample.env:16]: Let’s get rid of this diff

Comment (trel) [inline sample.env:16]: Let’s get rid of this diff - and then squash again.

Comment (ganning127) [reply sample.env:16]: I'm not really sure why this diff is here. You can view [my code here](https://github.com/ganning127/irods_client_zone_management_tool/blob/109.main.unix-space/sample.env) and the code on irods/main [here](https://github.com/irods/irods_client_zone_management_tool/blob/main/sample.env). They are the same---so not sure why the diff is there

Comment (trel) [reply sample.env:16]: i think the newline was removed.

Comment (ganning127) [reply sample.env]: aha! that's what it was

**Comment by trel** [comment]
Nice.   # it!

**Comment by ganning127** [comment]
added!

**Pull request merged**